maven_install.json

```diff
@@ -0,0 +1,1041 @@
+{
+    "dependency_tree": {
+        "__AUTOGENERATED_FILE_DO_NOT_MODIFY_THIS_FILE_MANUALLY": 1325301319,
```

Comment:
why do we need a generated file? Why are not the sha256 in the workspace? This looks to me worse than what we have.

Reply:
The idea is fairly similar to `npm`'s `package.json.lock` file. We keep the Maven transitive closure separate from Bazel in a lock file, and use the lock file to generate repository rules for each artifact on the fly. The underlying repository rules then become an implementation detail to the user. The only interface users have to manage Maven dependencies is the `maven_install` declaration in the WORKSPACE, which is simple and not tedious to maintain. 

More information: https://github.com/bazelbuild/rules_jvm_external#pinning-artifacts-and-integration-with-bazels-downloader